doc/source/examples/extended_examples/sfem/sfem.py

```diff
@@ -0,0 +1,758 @@
+# Copyright (C) 2016 - 2025 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import math
+import random
+from typing import Callable, Tuple
+
+import numpy as np
+
+
+def find_solution(
+    func: Callable[[float], float],
+    derivative_func: Callable[[float], float],
+    acceptable_solution_error: float,
+    solution_range: Tuple[float, float],
+) -> float:
+    """Find the solution of g(x) = 0 within solution range where g(x) is non-linear.
+
+    Parameters
+    ----------
+    func : Callable[float, float]
+        The function definition
```

Comment:
```suggestion
        Definition of the function.
```
We decided at the beginning of PyAnsys to drop the leading article off the brief description, primarily so that the first word was an important one. There should also be concluding punctuation.